server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/AbstractDocumentsFileManager.java

```diff
@@ -232,6 +243,14 @@ private long readMetrics(RandomAccessInput input, long offset, int numMetrics, O
                     metrics[i] = val;
                 }
                 offset += Long.BYTES;
+            } else if (aggregatedValueType instanceof CompensatedSumType) {
+                long val = input.readLong(offset);
+                if (isAggregatedDoc) {
+                    metrics[i] = new CompensatedSum(DOUBLE.toDoubleValue(val), 0);
```

Comment:
Can we use `metrics[i] = new CompensatedSum(aggregatedValueType.toDoubleValue(val), 0);`